portal-impl/src/com/liferay/portal/service/impl/CompanyLocalServiceImpl.java

```diff
@@ -1652,7 +1652,9 @@ else if (virtualHostname.equals(_DEFAULT_VIRTUAL_HOST) &&
 
 			throw new CompanyVirtualHostException();
 		}
-		else if (!Validator.isDomain(virtualHostname)) {
+		else if (!Validator.isDomain(virtualHostname) &&
```

Comment:
@stsquared99 According to your comment, a domain can be an IPV6 address, then what about IPV4 address? why we do not need to check IPV4 address?

Reply:
@tinatian Ah, I think that commit message should read "A host can be an IPv6 address" (I'm getting the terms mixed up even as I work on it). Without changing our current validators, isDomain is valid for domain names (www.url.com) and IPv4 addresses.

Reply:
@stsquared99 I see...
This is really hard for people to figure out why we only have checking for IPV6. I think we need to clean up the API before fixing this issue if you really want to depend on the Validator.


Tina.

Reply:
@tinatian 
I'm perfectly fine with that. This isn't a high priority at the moment, since IPv6 is not as common (unless someone tells us otherwise), so I'm ok with putting this on hold until we clean up the validators. Then we can revisit and decide how we want to deal with the IPv6 compression.

Thanks!